model/src/main/scala/explore/optics/Adjuster.scala

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.optics
+
+import cats.Contravariant
+import cats.Functor
+import cats.arrow.Choice
+import cats.arrow.Profunctor
+import cats.syntax.either._
+import monocle._
+
+// An Adjuster is just like a Setter, but weaker: it does not abide by the "compose modify" law.
+// It allows the modify/set functions to adjust the result (eg: not going out of bounds), which breaks composability.
+abstract class PAdjuster[S, T, A, B] extends Serializable { self =>
```

Comment:
Monocle is going to use only 2 types in the future, basically T=S and B=A but feel free to use this encoding. Julien also changed the type names to make them more descriptive, so it would be like `Adjuster[From, To]` I think

Reply:
Yup, this class is mostly copied and adapted from Monocle 2, and I left it like that to preserve the current approach, but maybe it's a good idea to adopt the new mechanics.